lib-extra/src/main/java/com/diffplug/spotless/extra/config/EclipseConfiguration.java

```diff
@@ -0,0 +1,257 @@
+	public EclipseConfiguration(String formatterName, Provisioner jarProvisioner, String... supportedEclipseVersions) {
+		Objects.requireNonNull(formatterName, "formatterName");
+		Objects.requireNonNull(jarProvisioner, "jarProvisioner");
+		Objects.requireNonNull(supportedEclipseVersions, "supportedEclipseVersions");
+		if (supportedEclipseVersions.length < 1) {
+			throw new IllegalArgumentException("At lease one version must be allowed.");
+		}
+
+		Path relativeFormatterResourcePath = Paths.get(ECLIPSE_FORMATTER_RESOURCES, formatterName.replace(' ', '_'));
+		eclipseConfigContext = getClass().getClassLoader().getResource(relativeFormatterResourcePath.toString());
+		if (null == eclipseConfigContext) {
+			throw new IllegalArgumentException(String.format("Eclipse configuration context resource path '%s' not found.", relativeFormatterResourcePath));
+		}
+
+		this.supportedEclipseVersions = new TreeSet<SemanticVersion>();
+		for (String allowedVersion : supportedEclipseVersions) {
+			this.supportedEclipseVersions.add(new SemanticVersion(allowedVersion));
+		}
+		version = this.supportedEclipseVersions.last(); //Use latest version per default
+		defaultCoordinatesUrl = getDefaultCoordinatesUrl(eclipseConfigContext, version);
+		this.jarProvisioner = jarProvisioner;
+
+		coordinateAdaptations = new MavenCoordinates(); //Use default coordinates configured for version
+		settingsFiles = new ArrayList<File>(); //Use default preferences
+		state = null; //Create state only on demand too speed up initialization
+	}
+
+	/** Set Eclipse version */
+	public void setVersion(String versionOrUrl) {
+		try {
+			// For development purpose and patch delivery it is allowed
+			// to specify directly the URL of a M2 coordinates file.
+			defaultCoordinatesUrl = new URL(versionOrUrl);
+			// The default version is not changed.
+		} catch (MalformedURLException ignored) {
+			SemanticVersion newVersion = new SemanticVersion(versionOrUrl);
+			if (supportedEclipseVersions.contains(newVersion)) {
+				defaultCoordinatesUrl = getDefaultCoordinatesUrl(eclipseConfigContext, newVersion);
+				version = newVersion;
+			} else {
+				throw new UserArgumentException(versionOrUrl,
+						String.format("Version is not part of the supported versions '%s'.",
+								supportedEclipseVersions.stream().map(v -> v.toString()).collect(Collectors.joining(", "))));
+			}
+		}
+		invalidateState();
+	}
+
+	private static URL getDefaultCoordinatesUrl(URL eclipseConfigContext, SemanticVersion version) {
+		try {
+			Path filePath = Paths.get(eclipseConfigContext.getPath(), String.format("v%s.txt", version.toString()));
+			return new URL(eclipseConfigContext.getProtocol(), eclipseConfigContext.getHost(), filePath.toString());
+		} catch (MalformedURLException e) {
+			/*
+			 *  This exception must be prevented by the strict syntax checking of
+			 *  SemanticVersion and unit-tests of all allowed versions.
+			 */
+			throw new RuntimeException(e);
```

Comment:
I don't know @nedtwigg's stance on throwing plain `RuntimeException`s, but if it were me, I'd elect to throw a more specific exception: `IllegalArgumentException` if the user passed in an invalid URL or some other illegal argument, or `IllegalStateException` if it's either not the user's fault or is due to unexpected state in `this`. :)